src/deadline/houdini_submitter/python/deadline_cloud_for_houdini/submitter.py

```diff
@@ -278,44 +315,64 @@ def _get_job_template(rop: hou.Node) -> dict[str, Any]:
         # remove dependency info from name
         connected_node["name"] = connected_node["rop"]
         rop_steps = [connected_node]
-        ignore_input_nodes = "false"
+        ignore_input_nodes = False
     for node in rop_steps:
-        # init data
-        init_data_contents = []
-        init_data_contents.append("scene_file: '{{Param.HipFile}}'\n")
-        init_data_contents.append(f"render_node: '{node['rop']}'\n")
-        init_data_contents.append(f"version: {_get_houdini_version()}\n")
-        init_data_contents.append(f"ignore_input_nodes: {ignore_input_nodes}\n")
-        init_data_contents.append(f"wedgenum: '{node['wedgenum']}'\n")
-        init_data_contents.append(f"wedge_node: '{node['wedge_node']}'\n")
+        init_data = {
+            "scene_file": "{{Param.HipFile}}",
+            "render_node": node["rop"],
+            "version": _get_houdini_version(),
+            "ignore_input_nodes": ignore_input_nodes,
+            "wedgenum": node["wedgenum"],
+            "wedge_node": node["wedge_node"],
+        }
         init_data_attachment = {
             "name": "initData",
             "filename": "init-data.yaml",
             "type": "TEXT",
-            "data": "".join(init_data_contents),
+            # Convert to dict to YAML string using the prettier nested object format
+            "data": yaml.safe_dump(init_data, default_flow_style=False),
         }
-        # environments
+
         environments = get_houdini_environments(init_data_attachment)
-        # task run data
-        task_data_contents = []
-        task_data_contents.append(f"render_node: {node['rop']}\n")
-        task_data_contents.append("frame: {{Task.Param.Frame}}\n")
-        task_data_contents.append(f"ignore_input_nodes: {ignore_input_nodes}\n")
-        # step
-        frame_range = "{start}-{stop}:{step}".format(**node)
+
+        task_data_dict = {"render_node": node["rop"], "ignore_input_nodes": ignore_input_nodes}
+
+        if node["render_strategy"] == RenderStrategy.SEQUENTIAL:
```

Comment:
Would you mind refactoring this function to adhere to the sonarcloud guidance: https://sonarcloud.io/project/issues?resolved=false&pullRequest=149&id=aws-deadline_deadline-cloud-for-houdini&open=AY-WcNeZcKfnCYQ8pJrR

?

Reply:
Refactored this function to reduce the branches. SonarCloud has some new complaints about existing code that I touched in the refactor. I don't think I'll fix those in this PR.